View3D/Services/WsModelGeneratorService.cs

```diff
@@ -78,6 +81,69 @@ public void GenerateWsModel()
             }
         }
 
+        public void GenerateWsModelForWh2()
+        {
+            var res = MessageBox.Show("Generate ws model for warhammer2?", "", MessageBoxButton.YesNo);
+            if (res != MessageBoxResult.Yes) return;
+
+            try
+            {
+                if (_packFileService.GetEditablePack() == null)
+                {
+                    MessageBox.Show("No editable pack selected", "error");
+                    return;
+                }
+
+                var isAllVisible = SceneNodeHelper.AreAllNodesVisible(_editableMeshNode);
+                bool onlySaveVisible = false;
+                if (isAllVisible == false)
+                {
+                    if (MessageBox.Show("Only generate for visible nodes?", "", MessageBoxButton.YesNo) == MessageBoxResult.Yes)
+                        onlySaveVisible = true;
+                }
+
+                var modelFile = _editorViewModel.MainFile;
+                var modelFilePath = _packFileService.GetFullPath(modelFile);
+                var wsModelPath = Path.ChangeExtension(modelFilePath, ".wsmodel");
+
+                var wsModelData = GenerateWsModelForWh2(modelFilePath, onlySaveVisible);
+                var existingWsModelFile = _packFileService.FindFile(wsModelPath, _packFileService.GetEditablePack());
+                SaveHelper.Save(_packFileService, wsModelPath, existingWsModelFile, Encoding.UTF8.GetBytes(wsModelData));
+            }
+            catch (Exception e)
+            {
+                _logger.Here().Error("Error generating ws model - " + e.Message);
+                MessageBox.Show("Generation failed!");
+            }
+        }
+
+        string GenerateWsModelForWh2(string modelFilePath, bool onlyVisible)
```

Comment:
This function should be pretty much the same for wh2 and wh3. Need to keep code duplication low